tests/FakeItEasy.Specs/CallMatchingSpecs.cs

```diff
@@ -234,46 +239,64 @@ public interface IHaveAnOutParameter
 
         [Scenario]
         public static void RefParameter(
-            IHaveARefParameter subject)
+            IHaveARefParameter subject,
+            string constraintValue,
+            string value,
+            bool result)
         {
-            "establish"
+            "Given a fake"
                 .x(() => subject = A.Fake<IHaveARefParameter>());
 
-            "when matching a call with a ref parameter"
+            "And a call to a method with a ref parameter configured on this fake"
                 .x(() =>
                     {
-                        string refString = "a constraint string";
-                        A.CallTo(() => subject.CheckYourReferences(ref refString))
-                            .Returns(true);
+                        constraintValue = "a constraint string";
+                        A.CallTo(() => subject.CheckYourReferences(ref constraintValue)).Returns(true);
                     });
 
-            "it should match when ref parameter value matches"
+            "When I make a call to the configured method with the constraint string"
                 .x(() =>
                     {
-                        string refString = "a constraint string";
-
-                        subject.CheckYourReferences(ref refString)
-                            .Should().BeTrue();
+                        value = constraintValue;
+                        result = subject.CheckYourReferences(ref value);
                     });
 
-            "it should not match when ref parameter value does not match"
-                .x(() =>
-                    {
-                        string refString = "a different string";
+            "Then it should return the configured value"
+                .x(() => result.Should().BeTrue());
 
-                        subject.CheckYourReferences(ref refString)
-                            .Should().BeFalse();
-                    });
+            "And it should assign the constraint value to the ref parameter"
+                .x(() => value.Should().Be(constraintValue));

```

Comment:
> value was already constraintValue, from line 260

I know, but that's what the original spec was doing...

> Perhaps use a local temp on that line? Or reset value before line 261?

If I do that, it won't match the configured call